test/e2e/targets/awskinesis/main.go

```diff
@@ -0,0 +1,366 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awskinesis
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+	"path"
+
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/aws/aws-sdk-go/service/kinesis/kinesisiface"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/apps"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/aws"
+	e2ekinesis "github.com/triggermesh/triggermesh/test/e2e/framework/aws/kinesis"
+	e2ece "github.com/triggermesh/triggermesh/test/e2e/framework/cloudevents"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/ducktypes"
+)
+
+/* This test suite requires:
+
+   - AWS credentials in whichever form (https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Sessions_options_from_Shared_Config)
+   - The name of an AWS region exported in the environment as AWS_REGION
+*/
+
+var targetAPIVersion = schema.GroupVersion{
+	Group:   "targets.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	targetKind     = "AWSKinesisTarget"
+	targetResource = "awskinesistargets"
+)
+
+const awsAccessKeyIDSecretKey = "awsApiKey"
+const awsSecretAccessKeySecretKey = "awsApiSecret"
+
+var _ = Describe("AWS Kinesis target", func() {
+	f := framework.New("awskinesistarget")
+
+	var ns string
+
+	var trgtClient dynamic.ResourceInterface
+
+	var streamARN string
+	var streamName string
+	var awsCreds credentials.Value
+	var awsSecret *corev1.Secret
+
+	BeforeEach(func() {
+		ns = f.UniqueName
+
+		gvr := targetAPIVersion.WithResource(targetResource)
+		trgtClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+	})
+
+	Context("a target is deployed", func() {
+		var trgtURL *url.URL
+		var kc kinesisiface.KinesisAPI
+
+		BeforeEach(func() {
+			sess := session.Must(session.NewSession())
+			kc = kinesis.New(sess)
+			awsCreds = readAWSCredentials(sess)
+
+			By("creating a Kinesis data stream", func() {
+				streamARN = e2ekinesis.CreateDatastream(kc, f)
+				streamName = path.Base(aws.ParseARN(streamARN).Resource)
```

Comment:
If the only reason for this string parsing and mangling to exist is to shorten the "deleting ..." messages logged by Ginkgo, I think we could simply log the ARN.

Reply:
The streamName is used for deleteStream and getRecords func

Reply:
Oh, I missed that, thanks!